lib/rubygems/server.rb

```diff
@@ -657,7 +657,7 @@ def root(req, res)
       "only_one_executable" => true,
       "full_name" => "rubygems-#{Gem::VERSION}",
       "has_deps" => false,
-      "homepage" => "http://docs.rubygems.org/",
+      "homepage" => "http://guides.rubygems.org/",
```

Comment:
I've followed a similar process for blog.rubidus.com, which is hosted on GitHub Pages and was able to do so with the free tier.  If you'd like help with that, I'd happy to work with you on making that a thing.